File: 数据结构/线性表/数组/283.移动零.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        cnt = 0
        for i in range(len(nums)):
            if nums[i]:
                nums[i-cnt], nums[i] = nums[i], nums[i-cnt]
            else:
                cnt = cnt + 1
        

File: 数据结构/线性表/数组/283.移动零.java
class Solution {
    public void moveZeroes(int[] nums) {
        int cnt = 0;
        for(int i=0;i<nums.length;i++){
            if(nums[i]==0){
                cnt ++;
            }else{
                int tmp = nums[i];
                nums[i] = nums[i-cnt];
                nums[i-cnt] = tmp;
            }
        }
    }
}